src/SwapAmounts/BlockIntervalDutchAuctionAmount.sol

```diff
@@ -0,0 +1,131 @@
+   * more favorable to the intent signer, and ends on a price that is more favorable to the solver.
+   
+   * In the case of "output", this would be a decreasing required amount. In the case of "input", this would be an increasing
+   * allowed amount.
+   *
+   * "auction" in variable names refers to the linear graph when amount is changing from startPercentE6 to endPercentE6. The settlement
+   * of a swap could occur before or after this range.
+   *
+   * Expects these encoded bytes params:
+   * - blockIntervalId: The id provided to the blockInterval segment
+   * - firstAuctionStartBlock: The block when the first dutch auction will start
+   * - auctionDelayBlocks: The number of blocks to wait for the next auction to start after the previous auction has closed. Should be
+   *                       set to the same value as intervalMinSize of blockInterval
+   * - auctionDurationBlocks: The number of blocks for amount to change from the start percent to the end percent relative to oracle
+   * - startPercentE6: Percentage of the PriceOracle reported amount where the auction curve should start, multiplied by 10**6
+   * - endPercentE6: Percentage of the PriceOracle reported amount where the auction curve should end, multiplied by 10**6
+   * - priceX96Oracle: IUint256Oracle that should report the price
+   * - priceX96OracleParams: params for the priceX96Oracle.getUint256() call
+   *
+   * When used with the blockInterval segment, getAmount() returns a dynamic value starting at a percentage above or below
+   * a reported price value, and ending at a percentage above or below a reported price value. 
+   * 
+   */
+  function getAmount (bytes memory params) public view returns (uint amount) {
+    (
+      uint oppositeTokenAmount,
+      uint64 blockIntervalId,
+      uint128 firstAuctionStartBlock,
+      uint128 auctionDelayBlocks,
+      uint128 auctionDurationBlocks,
+      int24 startPercentE6,
+      int24 endPercentE6,
+      address priceX96Oracle,
+      bytes memory priceX96OracleParams
+    ) = abi.decode(params, (uint, uint64, uint128, uint128, uint128, int24, int24, address, bytes));
+
+    // get the previous auction filled block from the block interval state. blockInterval segment sets this when the intent is filled
+    (uint128 previousAuctionFilledBlock,) = getBlockIntervalState(blockIntervalId);
+
+    // get the oracle price. Price is expected to be multiplied by 2**96.
+    uint priceX96 = IUint256Oracle(priceX96Oracle).getUint256(priceX96OracleParams);
+  
+    amount = getAuctionAmount(
+      uint128(block.number),
+      previousAuctionFilledBlock,
+      oppositeTokenAmount,
+      firstAuctionStartBlock,
+      auctionDelayBlocks,
+      auctionDurationBlocks,
+      startPercentE6,
+      endPercentE6,
+      priceX96
+    );
+  }
+
+  // a pure function to compute the auction amount based on input parameters
+  function getAuctionAmount (
```

Comment:
I think this will revert without a reason if auctionDurationBlocks is set to 0, since you divide by that at one point

Reply:
@Kyrrui I didn't see anywhere where this param is used in division. I might opt not to check for 0 values because technically it is valid to have a 0 block duration auction, it would just go immediately from startPercent to endPercent.

Reply:
@Kyrrui I tried to test it, it can't happen because this condition https://github.com/brinktrade/brink-intents/pull/43/files#diff-cb19e15413de8c208d926d8f98e311794b79ffdbdb33a6babbdb8802d29add6cR99 will never be executed if auctionStartBlock and auctionEndBlock are the same. So setting auction duration to zero does work as expected.